### PR TITLE
Adds new test for harvest source spatial format

### PR DESCRIPTION
**Pull request by btylerburton**
Adds new dataset to support the above
Swaps out CSW dataset to improve speed
Pins version of cypress interactive to same version in CLI
Cleans up tests before/after to not persist login across files
Updates dependencies and pins new geodatagov version

### CODE REVIEW

File: e2e/cypress.json [hunk 1]
```diff
@@ -1,8 +1,4 @@
 {
     "baseUrl": "http://ckan:5000",
-    "reporter": "junit",
-    "reporterOptions": {
-        "mochaFile": "cypress/results/output.xml"
```
Comment (nickumia-reisys) [inline e2e/cypress.json:5]: Is there any reason we don't want this output file anymore?  If we're worried about not seeing results in the terminal, we could add the `toConsole` parameter as well: https://docs.cypress.io/guides/tooling/reporters#Cypress-configuration-2

File: e2e/cypress/integration/ckan_extensions.spec.js [hunk 2]
```diff
@@ -23,14 +25,18 @@ describe('CKAN Extensions', () => {
             expect(installed_extensions).to.include('geodatagov_miscs');
             expect(installed_extensions).to.include('z3950_harvester');
             expect(installed_extensions).to.include('arcgis_harvester');
-            expect(installed_extensions).to.include('geodatagov_geoportal_harvester');
+            expect(installed_extensions).to.include(
+                'geodatagov_geoportal_harvester'
+            );
```
Comment (nickumia-reisys) [inline e2e/cypress/integration/ckan_extensions.spec.js:30]: Is there a line limit lint that put this on a new line?

File: e2e/cypress/integration/ckan_extensions.spec.js [hunk 3]
```diff
@@ -23,14 +25,18 @@ describe('CKAN Extensions', () => {
...
             expect(installed_extensions).to.include('waf_harvester_collection');
             expect(installed_extensions).to.include('geodatagov_csw_harvester');
             expect(installed_extensions).to.include('geodatagov_doc_harvester');
             expect(installed_extensions).to.include('geodatagov_waf_harvester');
             expect(installed_extensions).to.include('spatial_metadata');
             expect(installed_extensions).to.include('spatial_query');
-            expect(installed_extensions).to.include('spatial_harvest_metadata_api');
+            expect(installed_extensions).to.include(
+                'spatial_harvest_metadata_api'
+            );
```
Comment (nickumia-reisys) [inline e2e/cypress/integration/ckan_extensions.spec.js:39]: Same as last comment, expanding this seems trivially bad haha..

**Review by nickumia-reisys — APPROVED** [review]
Lots of good linting 🗃️

**Review by nickumia-reisys — DISMISSED** [review]
🗃️ 🗃️

File: e2e/cypress.json (hunk 1 above)
Comment (btylerburton) [reply e2e/cypress.json:5]: Oooh, good suggestion. Will rerun locally with those options and see if we get what we want.

File: e2e/cypress/integration/ckan_extensions.spec.js (hunk 2 above)
Comment (btylerburton) [reply e2e/cypress/integration/ckan_extensions.spec.js:30]: Yeah, this is Prettier's out of the box limit, which is set to 80 characters. I feel like I've seen a python limit somewhere, but can't recall where I've seen that. Do you know offhand?

Comment (nickumia-reisys) [reply e2e/cypress/integration/ckan_extensions.spec.js:30]: Yeah, the default python is also 80.  We set it to 128 in some of our repos, but yeah, I think somewhere around 120 should be a good limit.

File: e2e/cypress.json (hunk 1 above)
Comment (nickumia-reisys) [reply e2e/cypress.json:5]: No, probably not.  So it's okay to leave it as you changed it.  Just wanted to mention 😅

Comment (btylerburton) [reply e2e/cypress.json:5]: No worries. Always glad to spend a bit of effort to figure out what we really want. 

FYI current output to static output.xml only includes the last spec -- by design. https://docs.cypress.io/guides/tooling/reporters#Merging-reports-across-spec-files

File: e2e/cypress/integration/ckan_extensions.spec.js (hunk 2 above)
Comment (btylerburton) [reply e2e/cypress/integration/ckan_extensions.spec.js:30]: Changed my prettier config to 128 and repushing.

File: e2e/cypress.json (hunk 1 above)
Comment (nickumia-reisys) [reply e2e/cypress.json:5]: Good point, it wasn't entirely working before anyway haha..

**Comment by btylerburton** [comment]
just added prettierrc and had it lint the whole dir. @nickumia-reisys can i get another quick pass whenever it's convenient? thanks!

**Review by nickumia-reisys — APPROVED** [review]
🗃️ 🗃️ 🗃️

**Pull request merged**